recipe/build_opencv.sh

```diff
@@ -0,0 +1,150 @@
+#!/bin/bash
+set -ex
+
+# CMake FindPNG seems to look in libpng not libpng16
+# https://gitlab.kitware.com/cmake/cmake/blob/master/Modules/FindPNG.cmake#L55
+ln -s $PREFIX/include/libpng16 $PREFIX/include/libpng
+
+V4L="1"
+
+if [[ "${target_platform}" == linux-* ]]; then
+    # Looks like there's a bug in Opencv 3.2.0 for building with FFMPEG
+    # with GCC opencv/issues/8097
+    export CXXFLAGS="$CXXFLAGS -D__STDC_CONSTANT_MACROS"
+    OPENMP="-DWITH_OPENMP=1"
+fi
+
+if [[ "$build_variant" == "normal" ]]; then
+    echo "Building normal variant"
+    QT="5"
+else
+    echo "Building headless variant"
+    QT="0"
+    echo $QT
+fi
+
+if [[ "${target_platform}" == osx-* ]]; then
+    V4L="0"
+elif [[ "${target_platform}" == linux-ppc64le ]]; then
+    OPENVINO="0"
+fi
+
+
+if [[ "${target_platform}" != "${build_platform}" ]]; then
+    CMAKE_ARGS="${CMAKE_ARGS} -DProtobuf_PROTOC_EXECUTABLE=$BUILD_PREFIX/bin/protoc"
+    CMAKE_ARGS="${CMAKE_ARGS} -DQT_HOST_PATH=${BUILD_PREFIX}"
+fi
+
+
+export PKG_CONFIG_LIBDIR=$PREFIX/lib
+
+IS_PYPY=$(${PYTHON} -c "import platform; print(int(platform.python_implementation() == 'PyPy'))")
+
+LIB_PYTHON="${PREFIX}/lib/libpython${PY_VER}${SHLIB_EXT}"
+if [[ ${IS_PYPY} == "1" ]]; then
+    INC_PYTHON="$PREFIX/include/pypy${PY_VER}"
+else
+    INC_PYTHON="$PREFIX/include/python${PY_VER}"
+fi
+
+# FFMPEG building requires pkgconfig
+export PKG_CONFIG_PATH=$PKG_CONFIG_PATH:$PREFIX/lib/pkgconfig
+
+mkdir -p build${PY_VER}
+cd build${PY_VER}
+
+cmake -LAH -G "Ninja"                                                     \
```

Comment:
Some documentation of the choices of cmake options would be helpful

Reply:
Not sure what to put besides a link to the cmake file

Reply:
examples of non-obvious (to me) things : OPENCV_DOWNLOAD_TRIES, OPENCV_DOWNLOAD_PARAMS, ENABLE_PRECOMPILED_HEADERS, rationale for enabled / not enabled features (simply the maximum feature set that compiles? or was it just same as previous?)

Reply:
ok, would prefer some comments on the flags but won't block. It's not always the same rationale to select those flags so it's useful to know I think